File: patch.py

import re

with open("lib/features/loans/presentation/pages/loan_approval_page.dart", "r") as f:
    code = f.read()

if "dialog_utils.dart" not in code:
    code = code.replace("import 'package:flutter_bloc/flutter_bloc.dart';", "import 'package:flutter_bloc/flutter_bloc.dart';\nimport '../../../core/utils/dialog_utils.dart';")

code = re.sub(r"ScaffoldMessenger\.of\(context\)\.showSnackBar\([\s\S]*?Biometric signature required to accept agreement\.[\s\S]*?\);", "DialogUtils.showErrorDialog(context, 'Biometric signature required to accept agreement.');", code)
code = re.sub(r"ScaffoldMessenger\.of\(context\)\.showSnackBar\(const SnackBar\(content: Text\('Failed to initialize acceptance process'\)\)\);", "DialogUtils.showErrorDialog(context, 'Failed to initialize acceptance process.');", code)
code = re.sub(r"ScaffoldMessenger\.of\(context\)\.showSnackBar\([\s\S]*?Agreement accepted successfully![\s\S]*?\);", "DialogUtils.showSuccessDialog(context, 'Agreement accepted successfully!', onOk: () => context.pop(true));", code)
code = re.sub(r"ScaffoldMessenger\.of\(context\)\.showSnackBar\([\s\S]*?Failed to accept agreement\.[\s\S]*?\);", "DialogUtils.showErrorDialog(context, 'Failed to accept agreement. Please try again.');", code)

with open("lib/features/loans/presentation/pages/loan_approval_page.dart", "w") as f:
    f.write(code)

File: patch.js
const fs = require("fs");
let code = fs.readFileSync("lib/config/routes.dart", "utf8");
code = code.replace(
    "import '../features/auth/presentation/pages/mpin_setup_page.dart';",
    "import '../features/auth/presentation/pages/mpin_setup_page.dart';\nimport '../features/auth/presentation/pages/change_mpin_page.dart';"
);
code = code.replace(
    /GoRoute\(\s*path: AppConstants\.mpinSetup,\s*builder: \(context, state\) => const MpinSetupPage\(\),\s*\),/,
    `GoRoute(
      path: AppConstants.mpinSetup,
      builder: (context, state) => const MpinSetupPage(),
    ),
    GoRoute(
      path: "/change-mpin",
      builder: (context, state) => const ChangeMpinPage(),
    ),`
);
fs.writeFileSync("lib/config/routes.dart", code);

